engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
Hmmmm... the more I think about it the more I can come up with places where an instance of the WorldRenderer *should* be out of action and instead it might still be there, running even if it's not really useful. I.e., the CoreRegistry is one place where I don't think it gets removed from. There's also a WorldRendererSystem to handle chunk loading/unloading events and I don't know where/when that would get de-registered so that no references to it are left lying around. It contains a reference to the WorldRenderer and no checks for it to be null.